File: app-miyou/src/main/java/com/miyou/app/application/monitoring/port/MemoryExtractionMetricsPort.java
package com.miyou.app.application.monitoring.port;

public interface MemoryExtractionMetricsPort {
	void recordExtractionTriggered();

	void recordExtractionSuccess(int count);

	void recordExtractionFailure();

	void recordExtractedMemoryType(String type, int count);

	void recordExtractedImportance(double importance);
}

File: app-miyou/src/main/kotlin/com/miyou/app/application/monitoring/port/MemoryExtractionMetricsPort.kt
package com.miyou.app.application.monitoring.port

interface MemoryExtractionMetricsPort {
    fun recordExtractionTriggered()

    fun recordExtractionSuccess(count: Int)

    fun recordExtractionFailure()

    fun recordExtractedMemoryType(
        type: String,
        count: Int,
    )

    fun recordExtractedImportance(importance: Double)
}
